src/cata_utility.cpp

```diff
@@ -346,7 +348,20 @@ bool read_from_file( const std::string &path, const std::function<void( std::ist
         if( !fin ) {
             throw std::runtime_error( "opening file failed" );
         }
-        reader( fin );
+
+        // check if file is gzipped
+        // (byte1 == 0x1f) && (byte2 == 0x8b)
+        char header[2];
+        fin.read( header, 2 );
+        if( ( header[0] == '\x1f' ) && ( header[1] == '\x8b' ) ) {
+            const char *cstr_path = path.c_str();
+            igzstream fingz( cstr_path, std::ios::binary );
+            reader( fingz );
```

Comment:
`igzstream` always opens in binary mode (see `gzstreambuf::open`). You're passing some value here, and that overwrites default one that specifies the 'in' mode (see `igzstream` constructor), breaking the reading. Seems kinda weird to me to make it a default argument, but whatever

Curiously, fixing that wasn't enough. The game managed to load 2 compressed .map files, but got stuck on the 3rd one, entering some kind of infinite loop somewhere inside json error handling code (deep enough to confuse gdb). 

Changing the code to unpack the whole file into memory, and then read json from memory:
```suggestion
            igzstream fingz( path.c_str() );
            std::stringstream ss;
            ss << fingz.rdbuf();
            if( fingz.bad() ) {
                throw std::runtime_error( "reading file failed" );
            }
            ss.seekg( 0, std::ios::beg );
            reader( ss );
```
seems to be working, and the game loads compressed save successfully. Not sure, why, though. Maybe the reader uses some kind of seeking, and that doesn't work in igzstream?

Reply:
I was digging with `gdb` too and with `igzstream fingz( cstr_path );` `jsin->tell()` at `json.cpp:83` was returning `ffffffff`. Anyway, I'll try your code now, thanks again.